tools/fork-network/src/delayed_receipts.rs

```diff
@@ -0,0 +1,172 @@
+use crate::storage_mutator::ShardUpdateState;
+
+use near_chain::types::RuntimeAdapter;
+use near_crypto::PublicKey;
+use near_primitives::borsh;
+use near_primitives::receipt::{Receipt, ReceiptOrStateStoredReceipt, TrieQueueIndices};
+use near_primitives::shard_layout::{ShardLayout, ShardUId};
+use near_primitives::trie_key::TrieKey;
+use near_primitives::types::{ShardId, ShardIndex};
+use near_store::Trie;
+use nearcore::NightshadeRuntime;
+
+use anyhow::Context;
+use std::borrow::Cow;
+use std::collections::hash_map::Entry;
+use std::collections::{BTreeSet, HashMap};
+
+// Keeps track of the mapping of delayed receipts from a source shard to the
+// shards their receivers will belong to in the forked chain
```

Comment:
It is worth mentioning that we don't store Receipts directly due to RAM concerns. At least that's my impression.

Reply:
Oh yea that was why. Idk how important it will be in practice but I guess at least in theory it could get big. Actually now that I think about it, I should add a todo in `write_delayed_receipts()` because in that function we are storing everything in memory anyway :/ we should commit them periodically